bndtools.m2e/src/bndtools/m2e/MavenWorkspaceRepository.java

```
@@ -148,6 +131,63 @@ public IStatus run(IProgressMonitor monitor) {
+            try {
+                MavenUtil.execJarMojo(mavenProject, monitor);
+            } catch (CoreException e) {
+                failureReason = e.getMessage();
+            }
+        }
+
+        if (bundleFile.exists() && failureReason == null) {
+            try {
+                Domain domain = Domain.domain(bundleFile);
+                String bundleBsn = domain.getBundleSymbolicName().getKey();
+                Version bundleVersion = new Version(domain.getBundleVersion());
+
+                if (!bsn.equals(bundleBsn)) {
+                    failureReason = "Bundle built by maven changed bsn";
+                }
+
+                if (version.compareTo(bundleVersion) != 0) {
+                    failureReason = "Bundle built by maven changed version";
+                }
+            } catch (IOException e) {
+                failureReason = e.getMessage();
+            }
+        } else {
+            failureReason = "Maven build did not create bundle file";
+        }
+
+        return null;
+    }
+
+    private boolean isBundleFileOutOfDate(File bundleFile, MavenProject mavenProject) {
+        final File outputDirectory = new File(mavenProject.getBuild().getOutputDirectory());
+        final long[] outputDirectoryLastModified = new long[1];
+        outputDirectoryLastModified[0] = 0;
+
+        try {
+            Files.walkFileTree(outputDirectory.toPath(), new SimpleFileVisitor<Path>() {
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                    final long lastModified = file.toFile().lastModified();
+
+                    if (lastModified > outputDirectoryLastModified[0]) {
+                        outputDirectoryLastModified[0] = lastModified;
+                    }
+
+                    return super.visitFile(file, attrs);
+                }
+            });
+        } catch (IOException e) {}
```

I don't think we should discard the exception. At least log it.